MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,1771 @@
+      nsharedjets++;
+      std::vector<int>::iterator it = getFirstCommonElement(sv_match_tracktojet_which_temp_jetidx[vtx0idx], sv_match_tracktojet_which_temp_jetidx[vtx1idx]);
+      int it_idx = std::distance(sv_match_tracktojet_which_temp_jetidx[vtx0idx].begin(), it);
+      int jet_index = sv_match_tracktojet_which_temp_jetidx[vtx0idx].at(it_idx);
+
+      eraseElement(sv_match_tracktojet_which_temp_jetidx[vtx0idx], jet_index);
+      eraseElement(sv_match_tracktojet_which_temp_jetidx[vtx1idx], jet_index);
+
+      // start collecting shared tracks of sv0 for each shared jet
+      std::multimap<int, size_t> sv0_m;
+      for (size_t k = 0; k < sv0_match_tracktojet_which_jetidx.size(); k++)
+        if (sv0_match_tracktojet_which_jetidx[k] == jet_index) { sv0_m.insert({ sv0_match_tracktojet_which_jetidx[k], k }); }
+
+      for (auto it = sv0_m.begin(); it != sv0_m.end(); ) {
+        auto p = sv0_m.equal_range(it->first);
+
+        while (p.first != p.second) {
+          sv0_match_sharedjettotrack_which_temp_trkidx.push_back(sv0_match_tracktojet_which_trkidx[p.first++->second]);
+        }
+        it = p.second;
+      }
+
+      sv0_match_sharedjettotrack_which_trkidx.push_back(sv0_match_sharedjettotrack_which_temp_trkidx);
+
+      // start collecting shared tracks of sv1 for each shared jet
+      std::multimap<int, size_t> sv1_m;
+      for (size_t k = 0; k < sv1_match_tracktojet_which_jetidx.size(); k++) {
+        if (sv1_match_tracktojet_which_jetidx[k] == jet_index) { 
+          sv1_m.insert({ sv1_match_tracktojet_which_jetidx[k], k });
+        }
+      }
+
+      for (auto it = sv1_m.begin(); it != sv1_m.end(); ) {
+        auto p = sv1_m.equal_range(it->first);
+
+        while (p.first != p.second) {
+          sv1_match_sharedjettotrack_which_temp_trkidx.push_back(sv1_match_tracktojet_which_trkidx[p.first++->second]);
+        }
+        it = p.second;
+      }
+
+      sv1_match_sharedjettotrack_which_trkidx.push_back(sv1_match_sharedjettotrack_which_temp_trkidx);
+
+      if (sv0_match_sharedjettotrack_which_temp_trkidx.size() == 1) {
+        sv0_lonesharedtrack_which_trkidx.push_back(sv0_match_sharedjettotrack_which_temp_trkidx[0]);
+      }
+      if (sv1_match_sharedjettotrack_which_temp_trkidx.size() == 1) {
+        sv1_lonesharedtrack_which_trkidx.push_back(sv1_match_sharedjettotrack_which_temp_trkidx[0]);
+      }
+
+      sv0_match_sharedjettotrack_which_temp_trkidx = {};
+      sv1_match_sharedjettotrack_which_temp_trkidx = {};
+    }
+    sv_lonesharedtrack_which_trkidx.push_back(sv0_lonesharedtrack_which_trkidx);
+    sv_lonesharedtrack_which_trkidx.push_back(sv1_lonesharedtrack_which_trkidx);
+  }
+  bool shared_jet_specialcase = (sv0_lonesharedtrack_which_trkidx.size() != 0 || sv1_lonesharedtrack_which_trkidx.size() != 0);
+  return std::pair<bool, std::vector<std::vector<size_t>>>(
+      shared_jet_specialcase,
+      sv_lonesharedtrack_which_trkidx);
+}

```

Comment:
What here is the "special case"? I think this is just to track down the {1,n} (and {1,1}) shared jets? If this function only returns {1,n} shared jets, and never the {m,n} ones, maybe we should add a comment at the top of the `MFVVertexer::sharedjets` function.

Reply:
Done